security/permissions/src/main/resources/application.properties

```diff
@@ -28,3 +28,16 @@ quarkus.http.auth.policy.role-policy2.permissions.admin=read:minimal,read:all,cr
 quarkus.http.auth.permission.roles2.paths=/default-permission-allowed/custom-permission/*,/custom-annotation-permission-allowed/custom-permission/*
 quarkus.http.auth.policy.role-policy2.permission-class=io.quarkus.ts.openshift.security.basic.permissions.resources.helpers.CustomPermission
 quarkus.http.auth.permission.roles2.policy=role-policy2
+
+quarkus.http.auth.permission.roles3.shared=true
+quarkus.http.auth.permission.roles3.policy=role-policy3
+quarkus.http.auth.permission.roles3.paths=/bean-param/*
+quarkus.http.auth.policy.role-policy3.permissions.admin=read:basic,read:nested,read:detailed
```

Comment:
I think that it's probably enough for you to have `read` permission defined and there is no need for `read:basic`  and other as you don't check for this. You would need similar implementation as it's done in https://github.com/quarkusio/quarkus-security/blob/main/src/main/java/io/quarkus/security/StringPermission.java

So if I change in test the admin with user, they still pass as both of them have same permission (read). Maybe it would be better to change it for read and write or use only one "user".

Reply:
Good point, I've done some modifications to make it similar to StringPermission class, thanks.